src/upgrade.py

```diff
@@ -143,6 +158,14 @@ def _on_upgrade_granted(self, event: UpgradeGrantedEvent) -> None:
 
         self.charm._setup_exporter()
         self.charm.backup.start_stop_pgbackrest_service()
+        raft_encryption = (
+            int(
+                json.loads(self.peer_relation.data[self.charm.app].get("dependencies", "{}"))
+                .get("charm", {})
+                .get("version", 0)
+            )
+            < 3
+        )
```

Comment:
Check for versions that don't encrypt RAFT. Seems to be only an issue for 1 (current `14/stable`) but adding both previous version just in case.